rust/xaynet-client/src/mobile_client/client.rs

```diff
@@ -46,6 +57,27 @@ impl<Type> ClientState<Type> {
         warn!("reset client");
         ClientState::<Awaiting>::new(self.participant.reset(), self.round_params)
     }
+
+    async fn send_message<T: ApiClient>(
+        &self,
+        api: &mut T,
+        payload: Payload,
+    ) -> Result<(), ClientError<T::Error>> {
+        // Unwrapping is fine because this only errors out if the
+        // payload if a Chunk, which we never create in the client.
+        let encoder = MessageEncoder::<'_, Type>::new(
+            &self.participant,
+            payload,
+            self.round_params.pk,
+            MAX_PAYLOAD_SIZE,
+        )
+        .unwrap();
+        for part in encoder {
```

Comment:
Just for my understanding.
In the event that the client is terminated in the middle of the loop, the client would send the messages that are already received by the coordinator in the next attempt again + the missing ones? Can the coordinator already handle this case?

Reply:
>  the client would send the messages that are already received by the coordinator in the next attempt again + the missing ones?

Yes.

>  Can the coordinator already handle this 

Yes, it'll just ignore the parts if already has.